lib/galaxy/datatypes/tabular.py

```diff
@@ -1161,3 +1161,93 @@ def get_chunk(self, trans, dataset, chunk):
         ck_data_body = re.sub('[ ]+', '\t', ck_data_body)
 
         return dumps({'ck_data': util.unicodify(ck_data_header + "\n" + ck_data_body), 'ck_index': ck_index + 1})
+
+
+@build_sniff_from_prefix
+class MatrixMarket(Tabular):
+    """
+    The Matrix Market (MM) exchange formats provide a simple mechanism
+    to facilitate the exchange of matrix data. MM coordinate format is
+    suitable for representing sparse matrices. Only nonzero entries need
+    be encoded, and the coordinates of each are given explicitly.
+
+    The tabular file format is defined as follows::
+
+    %%MatrixMarket matrix coordinate real general <--- header line
+    %                                             <--+
+    % comments                                       |-- 0 or more comment lines
+    %                                             <--+
+        M  N  L                                   <--- rows, columns, entries
+        I1  J1  A(I1, J1)                         <--+
+        I2  J2  A(I2, J2)                            |
+        I3  J3  A(I3, J3)                            |-- L lines
+            . . .                                    |
+        IL JL  A(IL, JL)                          <--+
+
+    Indices are 1-based, i.e. A(1,1) is the first element.
+
+    >>> from galaxy.datatypes.sniff import get_test_fname
+    >>> MatrixMarket().sniff( get_test_fname( 'sequence.maf' ) )
+    False
+    >>> MatrixMarket().sniff( get_test_fname( '1.mtx' ) )
+    True
+    """
+    file_ext = "mtx"
+
+    def __init__(self, **kwd):
+        super(MatrixMarket, self).__init__(**kwd)
+
+    def sniff_prefix(self, file_prefix):
+        count = 0
+        nonzeros = 0
+        handle = file_prefix.string_io()
+        lines = handle.readlines()
+
+        if not lines[0].startswith('%%MatrixMarket matrix coordinate real general'):
+            return False
+
+        for line in lines:
+            if not line.startswith('%'):
+                split_line = line.split('\t')
+                if len(split_line) != 3:
+                    print("not 3 columns")
```

Comment:
this should be removed

Reply:
I removed it. 
Thanks for spotting it (remainder of local tests 😄)